Pass correct node bounds to PushOpacityMask.

grokys opened the pull request:
Use node bounds not clip bounds for opacity mask rect.

The opacity mask on the `Rectangle` on `CanvasPage` in ControlCatalog is now rendered correctly.

Fixes #1118

grokys committed:
Pass correct node bounds.

Use node bounds not clip bounds for opacity mask rect.

Fixes #1118.

nc4rrillo commented:
Also fixes #1693

ncarrillo reviewed (approved):
LGTM

Commit:
Merge branch 'master' into fixes/1118-opacity-mask

jmacato commented:
CI didnt liked this very much, it seems
```
Failed: Avalonia.Visuals.UnitTests.Rendering.DeferredRendererTests.Should_Push_Opacity_Mask
```

grokys committed:
Use layout bounds for opacity mask.

And added a new render test to make sure that the render transform is applied to it.

grokys committed:
Fix bad XAML formatting.

grokys commented:
Fixed the tests, now all passing.

ncarrillo committed:
Merge branch 'master' into fixes/1118-opacity-mask
